test-data/unit/check-modules.test

```diff
@@ -811,3 +811,35 @@ from a import x
 [file a/__init__.py]
 x = 0
 [out]
+
+
+-- Test stability under import cycles
+-- ----------------------------------
+
+-- The two tests are identical except one main has 'import x' and the other 'import y'
```

Comment:
Suggest adding more motivation for this test. In particular, when would one of these fail.